crates/ruff/src/importer/mod.rs

```diff
@@ -220,3 +229,13 @@ impl<'a> Importer<'a> {
         Ok(Edit::range_replacement(state.to_string(), stmt.range()))
     }
 }
+
+enum Resolution {
+    /// The symbol is available for use.
+    Success(Edit, String),
+    /// The symbol is imported, but the import came after the current location.
+    LateBinding,
+    /// The symbol is imported, but in an incompatible context (e.g., in typing-only context, while
+    /// we're in a runtime context).
+    IncompatibleContext,
+}
```

Comment:
This is perhaps a strange pattern, but I was finding all the `Option<Result<...>>` stuff to be really messy.

Reply:
I like this a lot. It communicates the outcome much clearer. We could even go one step further and flatten the Option too, but I think that would make the usage more cumbersome.